Deprecate Xsnapsync-bft-enabled option , to be removed in future release (#7930)
hyperledger/besu: CHANGELOG.md

  - Smart-contract-based permissioning
  - Proof of Work consensus
  - Fast Sync

> - Option `--Xsnapsync-bft-enabled` is deprecated and will be removed in a future release, enable snap sync with IBFT/QBFT permissioned Bonsai-DB chains by default

Reviewer: ```suggestion
- Option `--Xsnapsync-bft-enabled` is deprecated and will be removed in a future release. SNAP sync is supported for BFT networks
```
